### PR TITLE
Major code review

### PR DESCRIPTION
**Pull request by GabrielFleischer**
The PR aims at improving the code style.

Style issues and typos were addressed and a complete reformat was done

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MissionPicturesActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package ch.epfl.sdp.drone3d.ui.mission
```
Comment (jonasblanc) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/MissionPicturesActivityTest.kt:1]: Same remark as the other PR: missing header

File: app/src/main/java/ch/epfl/sdp/drone3d/service/api/drone/DroneService.kt [hunk 2]
```diff
@@ -4,6 +4,7 @@
  */
 
 package ch.epfl.sdp.drone3d.service.api.drone
+
 import io.mavsdk.System
 
 interface DroneService {
```
Comment (jonasblanc) [inline app/src/main/java/ch/epfl/sdp/drone3d/service/api/drone/DroneService.kt:10]: Interface documentation missing.

File: app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneServerFactoryImpl.kt [hunk 3]
```diff
@@ -8,11 +8,10 @@ package ch.epfl.sdp.drone3d.service.impl.drone
...
 import io.mavsdk.System
 import io.mavsdk.mavsdkserver.MavsdkServer
-import kotlinx.coroutines.*
-import org.jetbrains.annotations.Async
-import kotlin.concurrent.thread
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
 
-class DroneServerFactoryImpl: DroneServerFactory {
+class DroneServerFactoryImpl : DroneServerFactory {
```
Comment (jonasblanc) [inline app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneServerFactoryImpl.kt:14]: A documentation about the implemtation would be appreciated

File: app/src/main/java/ch/epfl/sdp/drone3d/service/impl/storage/FirebaseUser.kt [hunk 4]
```diff
@@ -8,33 +8,26 @@ package ch.epfl.sdp.drone3d.service.impl.storage
...
 import timber.log.Timber
+import javax.inject.Inject
 
 
-class FirebaseDatabase: Database {
-
-    private val database =
-        Firebase.database("https://drone3d-6819a-default-rtdb.europe-west1.firebasedatabase.app/")
-
+class FirebaseUser @Inject constructor(
```
Comment (jonasblanc) [inline app/src/main/java/ch/epfl/sdp/drone3d/service/impl/storage/FirebaseUser.kt:17]: A documentation about the class would be nice

File: app/src/main/java/ch/epfl/sdp/drone3d/ui/weather/WeatherInfoActivity.kt [hunk 5]
```diff
@@ -97,7 +97,7 @@ class WeatherInfoActivity : AppCompatActivity() {
         visibilityInfo.apply {
             val visibility = weatherReport.visibility
             text = if (visibility >= 1000) {
-                getString(R.string.info_visibility_km, (visibility/1000).toString())
+                getString(R.string.info_visibility_km, (visibility / 1000).toString())
```
Comment (jonasblanc) [inline app/src/main/java/ch/epfl/sdp/drone3d/ui/weather/WeatherInfoActivity.kt:100]: Maybe a constant for 1000 would improve the comprehension

**Review by jonasblanc — CHANGES_REQUESTED** [review]

**Review by jonasblanc — APPROVED** [review]

**Review by Anghul — APPROVED** [review]
Good job

**Pull request merged**